idle_sweep_test.go

```diff
@@ -66,6 +66,16 @@ func (pl *peerStatusListener) waitForZeroConnections(t testing.TB, channels ...*
 	}
 }
 
+func (pl *peerStatusListener) waitForZeroExchanges(t testing.TB, channels ...*Channel) {
+	for i := 0; i < 10; i++ {
+		if allExchangesEmpty(channels) {
+			return
+		}
+		<-time.After(100 * time.Millisecond)
```

Comment:
i'd suggest flipping the 10/100, so we run the loop 100 times, and wait 10ms in between checks. that way if the exchange clears right after we check, we won't have to wait 10ms, but 100ms.

also, since we're not using this in a channel, suggest doing `time.Sleep` here, and using `testutils.Timeout` to scale the number (could probably do `time.Sleep(testutils.Timeout(1ms))` and run that 100 times, which on CI is scaled up anyway